ecr/pusher.go

```diff
@@ -70,38 +75,79 @@ func (p ecrPusher) pushManifest(ctx context.Context, desc ocispec.Descriptor) (c
 			Error("ecr.pusher.manifest: failed to check existence")
 		return nil, err
 	}
+
+	// FIXME: confirm that the imageTag was also set as needed - its skipped
+	// otherwise (this is beside the atomic tag operation).
+	//
+	// If the goal was to accomplish the same as:
+	//
+	// docker push $ecr_uri:latest
+	// docker tag $ecr_uri:latest $ecr_uri:latest-amd64
+	// docker push $ecr_uri:latest-amd64
+	//
+	// Then this short circuited return will not cause the latest-amd64 tag to
+	// be written.
+
 	if exists {
 		log.G(ctx).Debug("ecr.pusher.manifest: content already on remote")
 		p.markStatusExists(ctx, desc)
 		return nil, errors.Wrapf(errdefs.ErrAlreadyExists, "content %v on remote", desc.Digest)
 	}
 
 	ref := p.markStatusStarted(ctx, desc)
+
+	var tag string
+	if p.isImageDescriptor(desc) {
+		tag, _ = p.ecrSpec.TagDigest()
+		log.G(ctx).WithField("tag", tag).Debug("ecr.pusher.manifest: including image tag in commit")
+	}
+
 	return &manifestWriter{
 		ctx:     ctx,
 		base:    &p.ecrBase,
 		desc:    desc,
 		tracker: p.tracker,
 		ref:     ref,
+		tag:     tag,
 	}, nil
 }
 
 func (p ecrPusher) checkManifestExistence(ctx context.Context, desc ocispec.Descriptor) (bool, error) {
-	image, err := p.getManifest(ctx)
+	// Fetch the manfests for this image.
+	images, err := p.getImages(ctx, *p.manifestImageIdentifier(desc))
 	if err != nil {
 		if err == errImageNotFound {
 			return false, nil
 		}
 		return false, err
 	}
-	if image == nil {
+	if len(images) < 1 {
 		return false, errors.New("ecr.pusher.manifest: unexpected nil image")
 	}
 
-	found := desc.Digest.String() == aws.StringValue(image.ImageId.ImageDigest)
+	found := desc.Digest.String() == aws.StringValue(images[0].ImageId.ImageDigest)
 	return found, nil
 }
 
+func (p ecrPusher) isImageDescriptor(desc ocispec.Descriptor) bool {
```

Comment:
nit: move below `manifestImageIdentifier`; caller above callee

Reply:
This is handled a bit differently in the latest set of changes.